File: Practica_10/src/Fila_B/Ministerio.java
package Fila_B;

public class Ministerio {
    private String nombre;
    private String direccion;
    private int nroEmpleados;
    private String[][] empleados = new String[3][100];
    private int[] edades = new int[100];
    private double[] sueldos = new double[100];
    // Constructor 1
    public Ministerio(String nombre, String direccion) {
        this.nombre = nombre;
        this.direccion = direccion;
        this.nroEmpleados = 0;
    }
    //Constructor 2 (por defecto)
    public Ministerio() {
        this("SinNombre", "SinDireccion");
    }
    //Agrega empleado
    public void agregarEmpleado(String nombre, int edad, double sueldo) {
        empleados[0][nroEmpleados] = nombre;
        empleados[1][nroEmpleados] = Integer.toString(edad);
        empleados[2][nroEmpleados] = Double.toString(sueldo);
        edades[nroEmpleados] = edad;
        sueldos[nroEmpleados] = sueldo;
        nroEmpleados++;
    }

    //(b) Eliminar empleados con edad X
    public void eliminarEmpleadoEdad(int edadX) {
        for (int i = 0; i < nroEmpleados; i++) {
            if (edades[i] == edadX) {
                for (int j = i; j < nroEmpleados - 1; j++) {
                    empleados[0][j] = empleados[0][j + 1];
                    empleados[1][j] = empleados[1][j + 1];
                    empleados[2][j] = empleados[2][j + 1];
                    edades[j] = edades[j + 1];
                    sueldos[j] = sueldos[j + 1];
                }
                nroEmpleados--;
                i--;
            }
        }
    }
    //(c) Transferir empleado con nombre X desde otro ministerio
    public void transferirEmpleadoDesde(Ministerio min, String nombreEmpleado) {
        for (int i = 0; i < min.nroEmpleados; i++) {
            if (min.empleados[0][i].equals(nombreEmpleado)) {
                int edad = min.edades[i];
                double sueldo = min.sueldos[i];
                this.agregarEmpleado(nombreEmpleado, edad, sueldo);
                min.eliminarPorIndice(i);
                break;
            }
        }
    }

    //Metodo auxiliar para eliminar por indice
    private void eliminarPorIndice(int indice) {
        for (int j = indice; j < nroEmpleados - 1; j++) {
            empleados[0][j] = empleados[0][j + 1];
            empleados[1][j] = empleados[1][j + 1];
            empleados[2][j] = empleados[2][j + 1];
            edades[j] = edades[j + 1];
            sueldos[j] = sueldos[j + 1];
        }
        nroEmpleados--;
    }

    //(d): Mostrar empleados con la menor edad
    public void mostrarMenorEdad() {
        if (nroEmpleados == 0) return;
        int menorEdad = edades[0];
        for (int i = 1; i < nroEmpleados; i++) {
            if (edades[i] < menorEdad) menorEdad = edades[i];
        }

        System.out.println("Empleados con menor edad (" + menorEdad + " años):");
        for (int i = 0; i < nroEmpleados; i++) {
            if (edades[i] == menorEdad) {
                System.out.println("- " + empleados[0][i]);
            }
        }
    }

    //(d) Mostrar empleados con el menor sueldo
    public void mostrarMenorSueldo() {
        if (nroEmpleados == 0) return;
        double menorSueldo = sueldos[0];
        for (int i = 1; i < nroEmpleados; i++) {
            if (sueldos[i] < menorSueldo) menorSueldo = sueldos[i];
        }

        System.out.println("Empleados con menor sueldo (" + menorSueldo + " Bs):");
        for (int i = 0; i < nroEmpleados; i++) {
            if (sueldos[i] == menorSueldo) {
                System.out.println("- " + empleados[0][i]);
            }
        }
    }

    public void mostrar() {
    	System.out.println("<MINISTERIO>");
        System.out.println("Ministerio: " + nombre + " | Direccion: " + direccion);
        System.out.println("Empleados: ");
        for (int i = 0; i < nroEmpleados; i++) {
            System.out.println(" " + empleados[0][i] + " | Edad: " + edades[i] + " | Sueldo: " + sueldos[i]);
        }
        System.out.println();
    }
}

File: Practica_10/src/Fila_B/Ministerio.py
class Ministerio:
    def __init__(self, nombre="SinNombre", direccion="SinDireccion"):
        self.nombre = nombre
        self.direccion = direccion
        self.nroEmpleados = 0
        self.empleados = [["" for _ in range(100)] for _ in range(3)] 
        self.edades = [0] * 100
        self.sueldos = [0.0] * 100

    # Agrega empleado
    def agregar_empleado(self, nombre, edad, sueldo):
        self.empleados[0][self.nroEmpleados] = nombre
        self.empleados[1][self.nroEmpleados] = str(edad)
        self.empleados[2][self.nroEmpleados] = str(sueldo)
        self.edades[self.nroEmpleados] = edad
        self.sueldos[self.nroEmpleados] = sueldo
        self.nroEmpleados += 1

    # (b) Eliminar empleados con edad X
    def eliminar_empleado_edad(self, edadX):
        i = 0
        while i < self.nroEmpleados:
            if self.edades[i] == edadX:
                for j in range(i, self.nroEmpleados - 1):
                    self.empleados[0][j] = self.empleados[0][j + 1]
                    self.empleados[1][j] = self.empleados[1][j + 1]
                    self.empleados[2][j] = self.empleados[2][j + 1]
                    self.edades[j] = self.edades[j + 1]
                    self.sueldos[j] = self.sueldos[j + 1]
                self.nroEmpleados -= 1
            else:
                i += 1

    # Metodo auxiliar para eliminar por índice
    def eliminar_por_indice(self, indice):
        for j in range(indice, self.nroEmpleados - 1):
            self.empleados[0][j] = self.empleados[0][j + 1]
            self.empleados[1][j] = self.empleados[1][j + 1]
            self.empleados[2][j] = self.empleados[2][j + 1]
            self.edades[j] = self.edades[j + 1]
            self.sueldos[j] = self.sueldos[j + 1]
        self.nroEmpleados -= 1

    # (c) Transferir empleado desde otro ministerio
    def transferir_empleado_desde(self, otro, nombre_empleado):
        for i in range(otro.nroEmpleados):
            if otro.empleados[0][i] == nombre_empleado:
                edad = otro.edades[i]
                sueldo = otro.sueldos[i]
                self.agregar_empleado(nombre_empleado, edad, sueldo)
                otro.eliminar_por_indice(i)
                break

    # (d.1) Mostrar empleados con menor edad
    def mostrar_menor_edad(self):
        if self.nroEmpleados == 0:
            return
        menor_edad = min(self.edades[:self.nroEmpleados])
        print(f"Empleados con menor edad ({menor_edad} años):")
        for i in range(self.nroEmpleados):
            if self.edades[i] == menor_edad:
                print(f"- {self.empleados[0][i]}")

    # (d.2) Mostrar empleados con menor sueldo
    def mostrar_menor_sueldo(self):
        if self.nroEmpleados == 0:
            return
        menor_sueldo = min(self.sueldos[:self.nroEmpleados])
        print(f"Empleados con menor sueldo ({menor_sueldo} Bs):")
        for i in range(self.nroEmpleados):
            if self.sueldos[i] == menor_sueldo:
                print(f"- {self.empleados[0][i]}")

    # Metodo para mostrar toda la informacion
    def mostrar(self):
        print("<MINISTERIO>")
        print(f"Ministerio: {self.nombre} | Direccion: {self.direccion}")
        print("Empleados:")
        for i in range(self.nroEmpleados):
            print(f" {self.empleados[0][i]} | Edad: {self.edades[i]} | Sueldo: {self.sueldos[i]}")
        print()

